modules/webgl/src/adapter/webgl-canvas-context.ts

```diff
@@ -52,6 +53,8 @@ export class WebGLCanvasContext extends CanvasContext {
    * See http://webgl2fundamentals.org/webgl/lessons/webgl-resizing-the-canvas.html
    */
   resize(options?: {width?: number; height?: number; useDevicePixels?: boolean | number}): void {
+    if (!this.device.gl) return;
```

Comment:
This should not be possible (i.e. a WebGLDevive should always have a gl context), unless this is some mid-initialization issue with function being called before Device is full constructed.

Reply:
Yes, is it due to the initialization sequence. Creating a `new WebGLCanvasContext` invokes `update()` and thus `resize()`, which expects there to be a `gl` context present on the device. Later the creation of the `gl` context can depend on a `canvasContext` already being available, so there is a circular dependency.

Merging this for now, and will try to followup with a better solution